performance_manager/lib/gtfs_utils.py

```diff
@@ -18,15 +16,32 @@ def add_event_hash_column(
         "start_date",
         "start_time",
         "vehicle_id",
-    )
+    ),
+) -> pandas.DataFrame:
+    """
+    provide consistent hash values for category columns of gtfs-rt events
+    """
+
     row_check = set(expected_hash_columns) - set(df_to_hash.columns)
     if len(row_check) > 0:
         raise IndexError(f"Dataframe is missing expected columns: {row_check}")
 
-    def apply_func(record: pandas.Series) -> int:
-        return hash(tuple(record[row] for row in expected_hash_columns))
+    # function to be used for hashing each record, 
+    # requires string as input returns raw bytes object
+    def apply_func(record: str) -> bytes:
+        return hashlib.md5(record.encode("utf8")).digest()
 
-    df_to_hash[hash_column_name] = df_to_hash.apply(apply_func, axis=1)
+    # vectorize apply_func so it can be used on numpy.ndarray object
+    vectorized_function = numpy.vectorize(apply_func)
+
+    # replace all "na" types values with python None to create consistent hash
+    df_to_hash = df_to_hash.fillna(numpy.nan).replace([numpy.nan], [None])
+
+    # convert rows of dataframe to concatenated string and apply vectorized
+    # hashing function
+    df_to_hash[hash_column_name] = vectorized_function(
+        df_to_hash[list(expected_hash_columns)].astype(str).values.sum(axis=1)
+    )
```

Comment:
completely re-did hashing behavior, python `hash` function has a random seed built in, changed to md5 hash on concatenated string of `expected_hash_columns`'s

this produces a bytes object that be directly stored in db field as binary field

Reply:
that looks good to me.